### ISSUE TITLE
Error in pathway_daa with ko abundances

### ISSUE THREAD

**Issue by andressamv** [cafferychen777/ggpicrust2#26]
Hi, I have another question, now about the function pathway_daa

I am running the following codes. While the first works normally, I receive the following error in the second:

daa_results <- pathway_daa(kegg_abundance, meta_table, 'Day', daa_method = 'ALDEx2', select = NULL,
                           p.adjust = 'bonferroni', reference = NULL)

daa_results2 <- pathway_daa(ko_abundance, meta_table, 'Day', daa_method = 'ALDEx2', select = NULL,
                              p.adjust = 'bonferroni', reference = NULL)

ALDEx2 takes a long time to complete the calculation, please wait patiently.
operating in serial mode
computing center with all features
operating in serial mode
Error in `contrasts<-`(`*tmp*`, value = contr.funs[1 + isOF[nn]]) : 
  contrasts can be applied only to factors with 2 or more levels

I am confused about this error, since the only difference between the codes is the first dataframe. Can you please help me with this too?

**Comment by cafferychen777** [comment]
Hi,

Thank you for reaching out. Regarding the error you received with the
second code block, it seems that there might be an issue with the data
structure of ko_abundance. It should be a data frame with row names as "ko"
and column names as "sample".

I would suggest checking the data structure of ko_abundance to ensure that
it is indeed a data frame with the correct row and column names. If the
data structure is correct, you may want to try running the code again to
see if the error persists.

Please let me know if you have any further questions or concerns.

Best,
Chen Yang






[image: Mailtrack]
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
Sender
notified by
Mailtrack
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
23/05/10
下午12:27:44

andressamv ***@***.***> 于2023年5月10日周三 06:59写道：

**Comment by andressamv** [comment]
Thanks! Yes, the error is still persisting.

My ko_abundance dataframe was created based on the pred_metagenome_unstrat.tsv file. The rownames are the kos (K00001, K00002, etc.) and the columns are the samples.

**Comment by cafferychen777** [comment]
Hi,

Based on the information you provided, it appears that the issue is related
to the ko_abundance dataframe that you created using the
pred_metagenome_unstrat.tsv file. Please note that this could be a problem
with the ALDEx2 package itself and unfortunately, there may not be a
straightforward solution to this issue.

In some cases, the same dataset may work on one computer but not on
another, which can be frustrating. However, I suggest that you try using
the LinDA method as an alternative. This is a very robust and widely
recognized method that you may find helpful in your analysis.

I hope this helps, and please let me know if you have any further questions
or concerns.

Best regards,
Chen YANG



andressamv ***@***.***> 于2023年5月10日周三 13:05写道：

**Comment by andressamv** [comment]
Thanks! In this case, I need to add a reference, and that would be a level of the 'group' variable?

I just tried with Deseq2, and now it is running well! I am just curious that I thought Deseq2 would do the reference vs everything else, but it seems that all comparisons were made.

**Comment by cafferychen777** [comment]
Yes, reference should be a level of "group" variable. The reason why DESEQ2
do the all comparisons is that I loop all the group combinationsin the code.




andressamv ***@***.***> 于2023年5月10日周三 13:50写道：

**Comment by andressamv** [comment]
great, thanks!
